src/dbcp/transform/fyi_queue.py

```diff
@@ -0,0 +1,321 @@
+    projects["power_market"] = projects["power_market"].replace({"ISO-NE": "ISONE"})
+
+    return projects
+
+
+def parse_capacity(s, n=3):
+    """Parse the capacity_by_generation_type_breakdown column into separate columns."""
+    if s == "nan":
+        return {}
+    try:
+        data = yaml.safe_load(s)
+    except Exception:
+        return {}
+    out = {}
+    for i, item in enumerate(data[:n], start=1):
+        out[f"resource_type_{i}"] = item.get("canonical_gen_type")
+        out[f"capacity_mw_resource_{i}"] = item.get("mw")
+    return out
+
+
+def _normalize_resource_capacity(fyi_df: pd.DataFrame) -> Dict[str, pd.DataFrame]:
+    """Pull out the awkward one-to-many columns (type_1, capacity_1, type_2, capacity_2) to a separate dataframe.
+
+    Args:
+        fyi_df (pd.DataFrame): FYI queue dataframe
+
+    Returns:
+        Dict[str, pd.DataFrame]: dict with the projects and multivalues split into two dataframes
+    """
+    # Apply parsing
+    parsed = fyi_df["capacity_by_generation_type_breakdown"].apply(parse_capacity)
+
+    # Expand into columns
+    parsed_df = pd.json_normalize(parsed).set_index(fyi_df.index)
+
+    # Join back
+    fyi_df = fyi_df.join(parsed_df)
+    n_multicolumns = 3
+    attr_columns = {
+        "resource": ["resource_type_" + str(n) for n in range(1, n_multicolumns + 1)],
+        "capacity_mw": [
+            "capacity_mw_resource_" + str(n) for n in range(1, n_multicolumns + 1)
+        ],
+    }
+    resource_capacity_df = normalize_multicolumns_to_rows(
+        fyi_df,
+        attribute_columns_dict=attr_columns,
+        preserve_original_names=False,
+        dropna=True,
+    )
+    combined_cols: List[str] = sum(attr_columns.values(), start=[])
+    project_df = fyi_df.drop(
+        columns=combined_cols + ["capacity_mw", "capacity_by_generation_type_breakdown"]
+    )
+
+    return {"resource_capacity_df": resource_capacity_df, "project_df": project_df}
+
+
+def _normalize_location(fyi_df: pd.DataFrame) -> Dict[str, pd.DataFrame]:
+    """Pull out the county and state columns to a separate dataframe.
+
+    Args:
+        fyi_df (pd.DataFrame): FYI queue dataframe
+
+    Returns:
+        Dict[str, pd.DataFrame]: dict with the projects and locations split into two dataframes
+    """
+    location_df = fyi_df[["raw_county_name", "raw_state_name"]]
+    location_df.dropna(
+        subset=["raw_state_name", "raw_county_name"], how="all", inplace=True
+    )
+    # reset project_id index
+    location_df = location_df.reset_index()
+    project_df = fyi_df.drop(columns=["raw_county_name", "raw_state_name"])
+    # TODO: temporarily drop, but really should investigate whether
+    # to just use fips_code column directly instead of geocoding.
+    # probably we want to just rename this column in the extraction
+    # to county_id_fips and pad it with 0's, then geocode for missing ones.
+    project_df = project_df.drop(columns=["fips_codes"])
```

Comment:
I dropped this column just to get everything to pass, but I think we should actually revisit this and see if we'd rather just use this `fips_codes` column from the raw data than geocoding for FIPS code. I'm sure there's some clean up to do on this column, and maybe we end up filling in with geocoding or validating with geocoding.

Reply:
I agree that it seems fine for now to use geocoding to fill in FIPS codes. I could see the `fips_codes` column being useful for filling in FIPS codes that the geocoding misses. I feel good about putting this in a follow up issue, along with more cleaning and analysis of location/FIPS columns in general.